pandas/_libs/parsers.pyx

```diff
@@ -279,73 +279,61 @@ cdef class TextReader:
 
     cdef public:
         int64_t leading_cols, table_width, skipfooter, buffer_lines
-        object allow_leading_cols
-        object delimiter, converters, delim_whitespace
+        bint allow_leading_cols, mangle_dupe_cols, memory_map, low_memory
+        bint delim_whitespace
+        object delimiter, converters
         object na_values
-        object memory_map
         object header, orig_header, names, header_start, header_end
         object index_col
-        object low_memory
         object skiprows
         object dtype
         object encoding
         object compression
-        object mangle_dupe_cols
         object usecols
         list dtype_cast_order
         set unnamed_cols
         set noconvert
 
     def __cinit__(self, source,
                   delimiter=b',',
-
                   header=0,
                   header_start=0,
                   header_end=0,
                   index_col=None,
                   names=None,
-
-                  memory_map=False,
+                  bint memory_map=False,
                   tokenize_chunksize=DEFAULT_CHUNKSIZE,
-                  delim_whitespace=False,
-
+                  bint delim_whitespace=False,
                   compression=None,
-
                   converters=None,
-
-                  skipinitialspace=False,
+                  bint skipinitialspace=False,
                   escapechar=None,
-                  doublequote=True,
+                  bint doublequote=True,
                   quotechar=b'"',
```

Comment:
bytes?

Reply:
When trying to have ```bytes quotechar=b'"'```, ```pytest``` is having a segfault.

---

```bash
$ pytest -q --cache-clear pandas/tests/io/parser/test_common.py
```

Outputs the following:

<details>

```
.Fatal Python error: Segmentation fault

Current thread 0x000064a291f35740 (most recent call first):
  File "/home/bummy/Documents/Github/Community/Python/Projects/pandas-MomIsBestFriend/pandas/io/parsers.py", line 1935 in __init__
  File "/home/bummy/Documents/Github/Community/Python/Projects/pandas-MomIsBestFriend/pandas/io/parsers.py", line 1112 in _make_engine
  File "/home/bummy/Documents/Github/Community/Python/Projects/pandas-MomIsBestFriend/pandas/io/parsers.py", line 880 in __init__
  File "/home/bummy/Documents/Github/Community/Python/Projects/pandas-MomIsBestFriend/pandas/io/parsers.py", line 449 in _read
  File "/home/bummy/Documents/Github/Community/Python/Projects/pandas-MomIsBestFriend/pandas/io/parsers.py", line 676 in parser_f
  File "/home/bummy/Documents/Github/Community/Python/Projects/pandas-MomIsBestFriend/pandas/tests/io/parser/conftest.py", line 22 in read_csv
  File "/home/bummy/Documents/Github/Community/Python/Projects/pandas-MomIsBestFriend/pandas/tests/io/parser/test_common.py", line 82 in test_empty_decimal_marker
  File "/home/bummy/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/_pytest/python.py", line 184 in pytest_pyfunc_call
  File "/home/bummy/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/pluggy/callers.py", line 187 in _multicall
  File "/home/bummy/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/pluggy/manager.py", line 86 in <lambda>
  File "/home/bummy/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/pluggy/manager.py", line 92 in _hookexec
  File "/home/bummy/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/pluggy/hooks.py", line 286 in __call__
  File "/home/bummy/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/_pytest/python.py", line 1479 in runtest
  File "/home/bummy/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/_pytest/runner.py", line 135 in pytest_runtest_call
  File "/home/bummy/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/pluggy/callers.py", line 187 in _multicall
  File "/home/bummy/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/pluggy/manager.py", line 86 in <lambda>
  File "/home/bummy/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/pluggy/manager.py", line 92 in _hookexec
  File "/home/bummy/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/pluggy/hooks.py", line 286 in __call__
  File "/home/bummy/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/_pytest/runner.py", line 217 in <lambda>
  File "/home/bummy/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/_pytest/runner.py", line 244 in from_call
  File "/home/bummy/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/_pytest/runner.py", line 217 in call_runtest_hook
  File "/home/bummy/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/_pytest/runner.py", line 186 in call_and_report
  File "/home/bummy/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/_pytest/runner.py", line 100 in runtestprotocol
  File "/home/bummy/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/_pytest/runner.py", line 85 in pytest_runtest_protocol
  File "/home/bummy/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/pluggy/callers.py", line 187 in _multicall
  File "/home/bummy/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/pluggy/manager.py", line 86 in <lambda>
  File "/home/bummy/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/pluggy/manager.py", line 92 in _hookexec
  File "/home/bummy/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/pluggy/hooks.py", line 286 in __call__
  File "/home/bummy/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/_pytest/main.py", line 272 in pytest_runtestloop
  File "/home/bummy/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/pluggy/callers.py", line 187 in _multicall
  File "/home/bummy/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/pluggy/manager.py", line 86 in <lambda>
  File "/home/bummy/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/pluggy/manager.py", line 92 in _hookexec
  File "/home/bummy/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/pluggy/hooks.py", line 286 in __call__
  File "/home/bummy/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/_pytest/main.py", line 247 in _main
  File "/home/bummy/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/_pytest/main.py", line 191 in wrap_session
  File "/home/bummy/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/_pytest/main.py", line 240 in pytest_cmdline_main
  File "/home/bummy/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/pluggy/callers.py", line 187 in _multicall
  File "/home/bummy/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/pluggy/manager.py", line 86 in <lambda>
  File "/home/bummy/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/pluggy/manager.py", line 92 in _hookexec
  File "/home/bummy/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/pluggy/hooks.py", line 286 in __call__
  File "/home/bummy/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/_pytest/config/__init__.py", line 125 in main
  File "/home/bummy/miniconda3/envs/pandas-dev/bin/pytest", line 11 in <module>
Segmentation fault (core dumped)

```

</details>

---

I believe that this has nothing to do with ```pytest``` but with ```python``` it self, because the first line in the error message is: ```Fatal Python error: Segmentation fault```